lib/sort_by_length.rb

```diff
@@ -1,7 +1,15 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  words = my_sentence.split(%r{\s+})
+  sorted_words = []
+  until words.length == 0
+    smallest_word = words.min_by { |w| w.length }
+    sorted_words.push(smallest_word)
+    words.delete(smallest_word)
```

Comment:
delete is an O(n) algorithm because it could cause each element to shift over one index.... This doesn't change the Big-O, but it does impact the coefficient.